docs/contribution/release.md

```diff
@@ -0,0 +1,78 @@
+# how release works
+
+Prerequisite:
+
+* docker installed
+* How drone works
+* How [goreleaser](https://goreleaser.com/intro/) works
+* multi arch TODO, don't know yet
+
+Briefly at least.
+
+## Continuous Delivery
+
+We use drone as continuous integration and continuous delivery pipeline.
+Integration means: running tests, checking code quality.
+Delivery means: make a release when a new tag get pushed.
+
+## goreleaser
+
+GoReleaser is a popular tool to release Go applications and library.
+
+I use it a lot because it is very flexible and it supports a lot of features, we
+need only a subset of them:
+
+* Multi arch build. It uses Go cross compilation feature to build for Darwin and
+  Linux (arm and amd)
+* Docker to build docker images for Linux container arm and amd.
+* Changelog to generate a changelog that will be pushed in the GitHub release
+  page with the list of PR part of the release itself.
+* Artifact (binaries) will be pushed as part of GitHub release page as well as
+  the other YAML file required by cluster-api like: metadata.yaml,
+  infrastructure-provider.yaml and so on.
+
+## Directories
+
+The release life cycle touches two directories:
+
+1. `./dist` it is a temporary directory used by gorleaser to bundle all the
+   required files and binaries as long as the generated changelog and so on. It
+   does not need to be checked in git and that's why it is in the gitignore.
```

Comment:
You don't need to add the last sentence, it can simply be, "dist is in gitignore."